### DISCUSSION TITLE
UI logging failure results in whole DAG failure

### DISCUSSION THREAD

**Discussion by j-adamczyk** [apache/airflow#18754] [General]
### Apache Airflow version

2.1.4 (latest released)

### Operating System

Alpine Linux 3.13

### Versions of Apache Airflow Providers

apache-airflow-providers-ftp==1.1.0
apache-airflow-providers-imap==1.0.1
apache-airflow-providers-sqlite==1.0.2

### Deployment

Docker-Compose

### Deployment details

Celery on Kubernetes, with logging on AWS S3.

### What happened

Tasks fail randomly (absolutely randomly, any task can fail with this exact message at any moment) with log:
```
*** Falling back to local log
*** Log file does not exist: /opt/airflow/logs/reporting_eventbrite_attendees/get_eventbrite_events/2021-10-03T00:00:00+00:00/1.log
*** Fetching from: http://:8793/log/reporting_eventbrite_attendees/get_eventbrite_events/2021-10-03T00:00:00+00:00/1.log
*** Failed to fetch log file from worker. The request to ':///' is missing either an 'http://'                         or 'https://' protocol.
```

This is caused by UI server fetching logs for the worker, not by worker itself (it does not show anything in the logs). So the whole DAG may fail due to Airflow UI problem, not due to any programmatic error.

### What you expected to happen

If the log is not found, just show "No logs could be fetched" or similar message. Mark task as success if no error was thrown in it. Continue the DAG execution.

### How to reproduce

This is not reproducible even in our case. This fails randomly, but for about ~1/3rd of the tasks. However, for all log fetching-related problems any UI problem (not only this particular one) should not result in task failure.

### Anything else

Problem appears randomly, with no apparent changes. There are no particular spikes in usage of CPU, RAM or anything else. Other DAGs that run concurrently with the failing task work properly.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
First of all description/title of the problem is wrong. There is no way UI error cause task failure, because there is no possibility of doing so. The logs suggest that you are mixing cause with resulr

This is not a reproducible case - and it looks like this is a deployment issue. It looks like some of your workers do not properly set their name properly - likely hostname-callable does not return the right host name https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#hostname-callable (which would indicate temporary DNS problem in your Kubernetes).

I'd recommend to follow: https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/

More information is needed from various logs (including logs of what is happening in your Kubernetes when name resolution happens) is needed in order to make it qualifies as Airflow issue. 

Switching to discussion unless more information is provided.
